runfolder/services.py

```diff
@@ -249,14 +276,63 @@ def _enumerate_runfolders(self):
                 directory = os.path.join(monitored_root, subdir)
                 self._logger.debug("Found potential runfolder {0}".format(directory))
                 state = self.get_runfolder_state(directory)
-                info = RunfolderInfo(self._host(), directory, state)
+                info = RunfolderInfo(self._host(), directory, state,
+                                     self.get_metadata(directory))
                 yield info
 
     def _requires_enabled(self, config_key):
         """Raises an ActionNotEnabled exception if the specified config value is false"""
         if not self._configuration_svc[config_key]:
             raise ActionNotEnabled("The action {0} is not enabled".format(config_key))
 
+    def get_metadata(self, path):
+        run_parameters = self.read_run_parameters(path)
+        reagent_kit_barcode = self.get_reagent_kit_barcode(path, run_parameters)
+        library_tube_barcode = self.get_library_tube_barcode(path, run_parameters)
+        metadata = {}
+        if reagent_kit_barcode:
+            metadata['reagent_kit_barcode'] = reagent_kit_barcode
+        if library_tube_barcode:
+            metadata['library_tube_barcode'] = library_tube_barcode
+        return metadata
+
+    def get_reagent_kit_barcode(self, path, run_parameters):
+        try:
+            barcode = run_parameters['RunParameters']['ReagentKitBarcode']
+        except KeyError:
+            # Reagent kit barcode is not available for all run types,
+            # it is therefore expected to not be found in all cases
+            self._logger.debug("Reagent kit barcode not found")
+            barcode = None
```

Comment:
Once again a minor thing, which does not matter very much in this case since it is such a simple function, but I would recommend doing `return None` directly in the exception clause. This makes the control flow easier to follow in more complicated scenarios.